### PR TITLE
Multiple values for filters

### PR DESCRIPTION
**Pull request by TheAnkurGoswami**
## What does this PR do?

Added support for multiple values for filter fields, i.e., array of values for a given filter field.

## Changes

* Using POST method for making API calls.
* Passing every parameter as request body.
* Token is sent via headers.

## Before reviewing

- [ ] Did you test this PR?
- [ ] Is there any migration or seeding necessary?
- [ ] If a new field is added to signals in the api, did you update the sample test files?

### Links:

- Jira: https://myaccern.atlassian.net/browse/RSL-481

### CODE REVIEW

File: packages/python/accern_data/accern_data.py [hunk 1]
```diff
@@ -435,7 +435,7 @@ def __init__(
             ) -> None:
         self._base_url = url
         self._token = token
-        self._filters: Dict[str, str] = {}
+        self._filters: Dict[str, Any] = {}
```
Comment (JosuaKrause) [inline packages/python/accern_data/accern_data.py:438]: create a type `FilterValue = Union[bool, int, str, List[str]]`

File: packages/python/accern_data/util.py [hunk 2]
```diff
@@ -52,7 +52,11 @@ def check_filters(
     return True
 
 
-def field_transformation(value: Any) -> str:
+def field_transformation(value: Any) -> Union[str, List[str]]:
+    if isinstance(value, list):
+        for idx, item in enumerate(value):
+            value[idx] = field_transformation(item)
```
Comment (JosuaKrause) [inline packages/python/accern_data/util.py:58]: can you have `List[List[...]]`? if not prevent validating that

File: packages/python/accern_data/accern_data.py (hunk 1 above)
Comment (TheAnkurGoswami) [reply packages/python/accern_data/accern_data.py:438]: `FilterValue = Union[bool, int, str, List[bool], List[int], List[str]]`

File: packages/python/accern_data/accern_data.py [hunk 3]
```diff
@@ -553,43 +546,30 @@ def get_last_silenced_errors(
...
-                pass
-        if date_type == "date":
-            date_type = "published_at"
-        return date_type, date_val
-
     def _read_total(
             self,
-            date: Dict[str, str],
-            filters: Dict[str, str],
+            params: Dict[str, str],
```
Comment (JosuaKrause) [inline packages/python/accern_data/accern_data.py:551]: what does params contain? at this point it should only be url_params -- all other params should go to json

File: packages/python/accern_data/accern_data.py [hunk 4]
```diff
@@ -617,23 +597,18 @@ def _read_date(
...
-                        harvested_after,
-                        params,
-                        mode.get_format(),
-                        filters=filters)
+                        params, mode.get_format(), filters=filters)
                 else:
                     req_params = {
-                        "token": self._token,
-                        **{**url_params, **filters, **params},
+                        **{**url_params, **params},
```
Comment (JosuaKrause) [inline packages/python/accern_data/accern_data.py:604]: params needs to go to json

Comment (JosuaKrause) [reply packages/python/accern_data/accern_data.py:604]: url_params need to go to query

Comment (JosuaKrause) [reply packages/python/accern_data/accern_data.py:604]: add json_params for explicit json parameters

File: packages/python/accern_data/accern_data.py [hunk 5]
```diff
@@ -553,43 +546,30 @@ def get_last_silenced_errors(
...
-                        date_type, date_val, filters)
+                    resp = get_overall_total_from_dummy(params, filters)
                 else:
                     req_params = {
-                        "token": self._token,
-                        **filters,
-                        **date,
+                        **params,
                         "format": "json",
                         "size": "1",
```
Comment (JosuaKrause) [inline packages/python/accern_data/accern_data.py:565]: this should be `"size": 1` then, no?

**Review by JosuaKrause — APPROVED** [review]
lgtm

**Review by JosuaKrause — APPROVED** [review]

**Pull request merged**